assets/scripts/aircraft.js

```diff
@@ -1327,17 +1328,31 @@ var Aircraft=Fiber.extend(function() {
           vector = vscale([sin(angle), cos(angle)], scaleSpeed);
         }
         else {
-          vector = vsum(vscale([sin(wind.angle + Math.PI), cos(wind.angle + Math.PI)],
+          var crab_angle = 0;
+
+          // Compensate for crosswind while tracking a fix or on ILS
+          if (this.requested.navmode == "fix" || this.mode == "landing")
+          {
+            var offset = angle_offset(this.heading, wind.angle + Math.PI);
+            crab_angle = Math.asin((wind.speed * Math.sin(offset)) /
+                                   this.speed);
+          }
+          vector = vsum(vscale([sin(wind.angle + Math.PI),
+                                cos(wind.angle + Math.PI)],
                                wind.speed * 0.000514444 * game_delta()),
-                        vscale([sin(angle), cos(angle)], scaleSpeed));
+                        vscale([sin(angle + crab_angle),
```

Comment:
I think you can just use vturn without a vector. It will turn a unitary vector `(1, 0)` to your angle then. Less typing and typo probability.